core-bundle/assets/backend.js

```diff
@@ -22,6 +22,23 @@ application.load(context.keys().map((key) => {
     }
 }).filter((value) => value));
 
+// BC layer to make elements using 'class="tl_chosen"' get a 'data-controller="contao--chosen"' attribute.
+new MutationObserver(function (mutationsList) {
+    for (const mutation of mutationsList) {
+        if (mutation.type === 'childList') {
+            mutation.addedNodes.forEach((element) => {
+                if (element.matches && element.matches('select.tl_chosen') && element.getAttribute('data-controller') !== 'contao--chosen') {
+                    element.setAttribute('data-controller', 'contao--chosen');
+                }
+            })
+        }
+    }
+}).observe(document, {
+    attributes: false,
+    childList: true,
+    subtree: true
+});
+
```

Comment:
Hm, can't we add the attribute via the data container output?

Reply:
We should do this, yes, but I thought it's very likely people are relying on the mechanism (just adding a class).

Reply:
Hm, I guess so, yeah.

Reply:
I replaced all occurrences where we output `tl_chosen` with the respective `data-controller` attribute in 28f00e7a795ec43e2679608b1a7c569871c6ba58. I think would be fine with removing the BC layer then. Wdyt?

Reply:
Yes, let‘s remove it then.

Reply:
Not sure. I also have projects where I simply added the `chosen` class. I mean it's fine for me as I will know to change this to `data-controller="contao--chosen"` when upgrading to 5.4, but I guess it's still a BC break?

Reply:
I removed it for now. We can still re-add it later on if we decide it'S necessary.